### ISSUE TITLE
The extension doesn't work with Expo Cli

### ISSUE THREAD

**Issue by The-Developeur** [Microsoft/vscode-react-native#834]
I have a problem,
Recently as you can see on the screenshot under create-react-native-native-app is now outdated.
![image](https://user-images.githubusercontent.com/25546627/48663456-bf459a00-ea90-11e8-8abf-27fca18f251a.png)
And the new name of the extension is expo-cli 
https://expo.io/

Can you support debuging expo cli app using visual studio code ?
Thanks

link of the create-react-native-native-app repo : https://github.com/react-community/create-react-native-app

**Comment by ruslan-bikkinin** [comment]
Hi @The-Developeur and thanks for asking. Extension already support debugging in Expo. Please refer to [this documentation section](https://github.com/Microsoft/vscode-react-native/blob/master/doc/expo.md#debugging-expo-applications) for more details. Also, be aware that `Configuring Expo` section is outdated, but we already proposed [a PR](https://github.com/Microsoft/vscode-react-native/pull/826) with update, so if you need more information about advanced configuration, please, turn to it for now.

Please let us know, if you have any questions.

**Comment by hatton** [comment]
My apologies if this isn't close enough to what this issue is about...

I created an app using ``expo 
I followed the instructions in the PR #826, ran the "Debug in Exponent", logged in to expo when prompted, scanned the QR code, runs the sample app on my phone, all good! However vscode does not stop at breakpoints, and after a few minutes I get an error, "**Launch/Request timeout**".

I note from [the instructions](https://github.com/Microsoft/vscode-react-native/blob/master/doc/expo.md) that I should expect some files to be created in .vscode/ , e.g. exponentIndex.js . These files are not created. All I see is .vscode/.react/debuggerWorker.js

My launch.json is

```
  "configurations": [
    {
      "name": "Debug in Exponent",
      "program": "${workspaceRoot}/.vscode/launchReactNative.js",
      "type": "reactnative",
      "request": "launch",
      "platform": "exponent",
      "sourceMaps": true,
      "outDir": "${workspaceRoot}/.vscode/.react"
    }
  ]
```

I'm looking for something obvious like in [this reply](https://github.com/Microsoft/vscode-react-native/issues/793#issuecomment-421038283) but not seeing it, sorry!

[github repo](https://github.com/hatton/brexpo)
[output log](https://pastebin.com/TJMm4q0d)
expo version 2.4.0
extension version 0.6.18
node 10.13.0-64

**Comment by ruslan-bikkinin** [comment]
@hatton I could debug your application running on Windows machine and Android emulator without any problem. 

> I'm looking for something obvious like in this reply but not seeing it, sorry!
So, did you try that solution? Did it help you?

Sorry if I confused you with my answer about the docs. The most recent documentation is in `improve-docs` branch and we are going to merge it soon. So for now, please turn to [this documentation file](https://github.com/Microsoft/vscode-react-native/blob/improve-docs/doc/expo.md) for getting more details about how to debug Expo apps until it merged.

**Comment by Morriz** [comment]
> Sorry if I confused you with my answer about the docs. The most recent documentation is in `improve-docs` branch and we are going to merge it soon. So for now, please turn to [this documentation file](https://github.com/Microsoft/vscode-react-native/blob/improve-docs/doc/expo.md) for getting more details about how to debug Expo apps until it merged.

That link is dead, but if I open the[ correct one ](https://github.com/Microsoft/vscode-react-native/blob/master/doc/expo.md#debugging-expo-applications)(mentioned above), and follow those instructions, I still get errors:

```
Debug adapter process has terminated unexpectedly (read error)
```

**Comment by ruslan-bikkinin** [comment]
@Morriz seems to be that you have completely different case, please, submit separate issue.
> That link is dead, 

Thanks for pointing on that, changes from that PR already in master.

@hatton do you still have a problems debugging Expo App? If not, i'd close this.

**Comment by Morriz** [comment]
I do. Also, your newly pointed documentation still refers to old expo stuff:
```
the extension will install xde and @expo/ngrok when this feature is used for the first time.
```

**Comment by Morriz** [comment]
And now it barks on `.vscode/exponentIndex.js` not found

**Comment by ruslan-bikkinin** [comment]
@Morriz please submit separate issue.

**Comment by Morriz** [comment]
Why, I am having the issue as described exactly in the title

**Comment by The-Developeur** [comment]
Thank you  @ruslan-bikkinin

**Comment by hatton** [comment]
>>    I'm looking for something obvious like in this reply but not seeing it, sorry!

> So, did you try that solution? Did it help you?

In that screenshot, there is a "Debug JS" link on the phone. But I don't see anything like that in the Android expo app. Similarly, [the instructions](https://github.com/Microsoft/vscode-react-native/blob/master/doc/expo.md#debugging-expo-applications) now show a different screenshot with that kind of button, but again, only a partial screenshot, so you can't see where in the app you should see this.

Here's what I see: https://i.imgur.com/PDxSEvg.png, https://i.imgur.com/6OG8FNO.png.

Also, do let me know if you'd like a different issue for @Morriz and I seeing no .vscode/exponentIndex.js.  Thanks!

**Comment by The-Developeur** [comment]
@ruslan-bikkinin  When i start my project with the "Debug in Exponent" configuration i have this error
![image](https://user-images.githubusercontent.com/25546627/49226596-64446900-f3e7-11e8-8621-08451eacb3c9.png)

And in my .vscode Folder i don'y have launchReactNative.js

I have an other error when i select on the expo app remote js debuging i have this error on the console
![image](https://user-images.githubusercontent.com/25546627/49226705-a9689b00-f3e7-11e8-927d-56ba7249b5be.png)

**Comment by hatton** [comment]
OK, I read more instructions, more carefully, and found that the secret to getting to the developer menu is literally shaking your device (that's a new one!). And now we have https://github.com/Microsoft/vscode-react-native/issues/846 for the lack of generated .vscode files, so my part of this issue is dealt with. Thanks for the help!

**Comment by The-Developeur** [comment]
@hatton  I hnow but when i select remote debuging expo said me i need chrome to allow me remote debuging

**Comment by ruslan-bikkinin** [comment]
@hatton thanks for your reply. Do you experience the same issue now like in #846?

@The-Developeur 
> And in my .vscode Folder i don'y have launchReactNative.js

Technically debugger doesn't require `.vscode/launchReactNative.js` and this file is no longer being  generated inside .vscode. This is undocumented yet but the thing is that it couldn't be a reason for your problem. 

What OS do you use? What platform are you targeting: android or iOS? Are you running your app on real device or simulator/emulator?

> @hatton I hnow but when i select remote debuging expo said me i need chrome to allow me remote debuging

Could you please show me a screenshot of that?

**Comment by The-Developeur** [comment]
### Actual Behavior
i start the debugger with the configuration Debug in Exponent 

![image](https://user-images.githubusercontent.com/25546627/49229179-47ab2f80-f3ed-11e8-9304-b1c5143ddf82.png)

I Have this error  
![image](https://user-images.githubusercontent.com/25546627/49229251-64476780-f3ed-11e8-9233-47c2275f953f.png)

The Debugger ShutDown

 ### Expected Behavior

  debugger to continue

### Software

  I use the expo app on ios



This is my error when i selected "use a remote debugger"

![image](https://user-images.githubusercontent.com/25546627/49229512-f8193380-f3ed-11e8-9174-286e0b411a0e.png)


![image](https://user-images.githubusercontent.com/25546627/49229542-0bc49a00-f3ee-11e8-82e3-9d1a7a2eaaa4.png)

**Comment by hatton** [comment]
> @hatton thanks for your reply. Do you experience the same issue now like in #846?

Yes, I still do not see .vscode/exponentIndex.js. 

![image](https://user-images.githubusercontent.com/8448/49250683-acaa5980-f3dc-11e8-85b3-86281be61956.png)

**Comment by ruslan-bikkinin** [comment]
@The-Developeur if you use `Debug In Exponent` configuration then you don't need to run `expo start`.
All you need to do is to run `Debug In Exponent` configuration, wait until QR code window appears, open Expo app on your device/simulator, scan or copy-paste qr code to your app, wait until app is loaded, open Developer Menu, enable remote js debugging and that's it. 

Regarding this error:
![image](https://user-images.githubusercontent.com/30265843/49277901-35380100-f494-11e8-9035-8e67732a38fb.png)

Could you please give us exact reproduction steps for it and these logs:
- Output of the Debug Console (View -> Toggle Debug Console):
```
PASTE OUTPUT HERE
```
- Output of the React-Native output channel (View -> Toggle Output -> Select React-Native in ListBox):
```
PASTE OUTPUT HERE
```
- Output of the Developer Tools console (Help -> Toggle Developer Tools -> Select Console tab):
```
PASTE OUTPUT HERE
```

Also, please be aware, that there are some issues with logging in current version of VS Code 1.29, but in latest [Insiders build](https://code.visualstudio.com/insiders/) these issues are resolved, so, please, use Insiders build for it.

**Comment by The-Developeur** [comment]
Output of the Debug Console (View -> Toggle Debug Console):

Nothing.
![image](https://user-images.githubusercontent.com/25546627/49300296-223c2580-f4c2-11e8-9c7f-f8beaa01c827.png)


    Output of the React-Native output channel (View -> Toggle Output -> Select React-Native in ListBox):

    I don't have react native in the Listbox 
![image](https://user-images.githubusercontent.com/25546627/49300456-924aab80-f4c2-11e8-9695-354b2ccba56a.png)

![image](https://user-images.githubusercontent.com/25546627/49300479-a098c780-f4c2-11e8-9ac1-50abe7fe7542.png)

    I have the extension;
![image](https://user-images.githubusercontent.com/25546627/49300508-b3ab9780-f4c2-11e8-813f-dc2a1477d49c.png)



    Output of the Developer Tools console (Help -> Toggle Developer Tools -> Select Console tab):

  

  | t.onDidNotificationChange | @ | notificationsAlerts.ts:40
-- | -- | -- | --
  | (anonymous) | @ | notificationsAlerts.ts:26
  | e.fire | @ | event.ts:145
  | t.notify | @ | notifications.ts:113
  | t.error | @ | notificationService.ts:48
  | (anonymous) | @ | debugService.ts:511
  | e.fire | @ | event.ts:145
  | (anonymous) | @ | debugSession.ts:749
  | e.fire | @ | event.ts:145
  | t.fireAdapterExitEvent | @ | rawDebugSession.ts:461
  | (anonymous) | @ | rawDebugSession.ts:440
  | _tickCallback | @ | internal/process/next_tick.js:188
  | Promise resolved (async) |   |  
  | t.stopAdapter | @ | rawDebugSession.ts:439
  | (anonymous) | @ | rawDebugSession.ts:427
  | Promise rejected (async) |   |  
  | t.shutdown | @ | rawDebugSession.ts:423
  | (anonymous) | @ | rawDebugSession.ts:80
  | e.fire | @ | event.ts:145
  | (anonymous) | @ | debugAdapter.ts:378
  | emitOne | @ | events.js:121
  | emit | @ | events.js:211
  | Socket._destroy.cb._handle.close | @ | net.js:554



![image](https://user-images.githubusercontent.com/25546627/49300602-ef466180-f4c2-11e8-8f93-51d1a3b50ada.png)

**Comment by The-Developeur** [comment]
@ruslan-bikkinin

**Comment by ruslan-bikkinin** [comment]
@The-Developeur it's still not clear, what cause your problem. Could you please, share with us demo application so we could try to reproduce it on our side?
Also, please fill these items:

## Software versions
- React-Native VS Code extension version: 
- VSCode version: 
- OS platform and version: 
- NodeJS version: 
- React Native version: 
- Expo SDK version (include if relevant):

**Comment by ruslan-bikkinin** [comment]
@The-Developeur do you have a chance to look at this, please?

**Comment by ruslan-bikkinin** [comment]
Hi @The-Developeur and sorry for delayed response. Your project works without any problems in my case. I  was able to reproduce your problem only once, when for some unknown reason all files except directrories were missing in node_modules/react-native. I'd recommend you to remove `node_modules` directory and run `npm install` again. Please let me know if it helps.

**Comment by The-Developeur** [comment]
Okay,
thank you for your helping ^^
Finally i use your extension with the react native cli and it work perfectly !
